src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -119,89 +148,1195 @@ static DamageTable D_808DBA44 = {
+        sp54.y += this->unk_380[i].y;
+        sp54.z += this->unk_380[i].z;
+        Math_Vec3f_Copy(&this->unk_380[i], &sp54);
+    }
+    func_800BF7CC(globalCtx, &this->actor, this->unk_380, ARRAY_COUNT(this->unk_380), 3, 0.1f, 0.3f);
+}
+
+void func_808D8FC4(EnSw* this, GlobalContext* globalCtx) {
+    Math_Vec3f_ToVec3s(&this->collider.dim.worldSphere.center, &this->actor.world.pos);
+    this->collider.dim.worldSphere.radius = this->collider.dim.modelSphere.radius * this->collider.dim.scale;
+    if (!(this->unk_410 & 0x10) && (this->actor.colChkInfo.health != 0) && (DECR(this->unk_458) <= 10)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+
+    if (!(this->unk_410 & 0x10) && (this->actor.colChkInfo.health != 0)) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+
+    if (this->unk_410 & 4) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-extern ColliderSphereInit D_808DB9E0;
-extern CollisionCheckInfoInit2 D_808DBA0C;
-extern DamageTable D_808DBA18;
-extern CollisionCheckInfoInit2 D_808DBA38;
-extern DamageTable D_808DBA44;
+s32 func_808D90C4(EnSw* this) {
+    if (this->actor.colChkInfo.damage < this->actor.colChkInfo.health) {
+        this->actor.colChkInfo.health -= this->actor.colChkInfo.damage;
+    } else {
+        this->actor.colChkInfo.health = 0;
+    }
+    return this->actor.colChkInfo.health;
+}
 
-extern UNK_TYPE D_06004788;
-extern UNK_TYPE D_06005298;
+void func_808D90F0(EnSw* this, s32 arg1, s16 arg2) {
+    Vec3f sp2C;
+    s16 sp2A;
+    s16 temp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sw/func_808D8940.s")
+    if (arg1 == 1) {
+        sp2A = this->unk_494;
+        Math_ApproachS(&this->unk_494, this->unk_496, 4, 0xE38);
+        temp = this->unk_494 - sp2A;
+    } else {
+        temp = arg2;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sw/func_808D8B58.s")
+    SysMatrix_InsertRotationAroundUnitVector_f(BINANG_TO_RAD(temp), &this->unk_368, 0);
+    SysMatrix_MultiplyVector3fByState(&this->unk_350, &sp2C);
+    Math_Vec3f_Copy(&this->unk_350, &sp2C);
+    Math3D_CrossProduct(&this->unk_368, &this->unk_350, &this->unk_35C);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sw/func_808D8D60.s")
+s32 func_808D91C4(EnSw* this, CollisionPoly* arg1) {
+    f32 sp4C;
+    f32 temp_f12;
+    f32 temp_f0;
+    Vec3f sp38;
+    Vec3f sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sw/func_808D8ED0.s")
+    this->actor.floorPoly = arg1;
+    if (arg1 != 0) {
+        sp38.x = COLPOLY_GET_NORMAL(arg1->normal.x);
+        sp38.y = COLPOLY_GET_NORMAL(arg1->normal.y);
+        sp38.z = COLPOLY_GET_NORMAL(arg1->normal.z);
+    } else {
+        return false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sw/func_808D8FC4.s")
+    temp_f12 = DOTXYZ(sp38, this->unk_368.x);
```

Comment:
```suggestion
    temp_f12 = DOTXYZ(sp38, this->unk_368);
```
This could be a reason why jenkins isn't happy